src/bin/rustfmt.rs

```diff
@@ -77,6 +77,19 @@ fn lookup_and_read_project_file(input_file: &Path) -> io::Result<(PathBuf, Strin
     Ok((path, toml))
 }
 
+/// read the given config file path recursively if present else read the project file path
+fn match_config_path_or_input_file(input_file: &Path,
+                                   config_path: Option<&String>)
+                                   -> io::Result<(PathBuf, String)> {
+    match config_path {
+        Some(config_file) => {
+            lookup_and_read_project_file(config_file.as_ref())
+                .or(lookup_and_read_project_file(input_file))
```

Comment:
Could make this an `.or_else(|_| lookup_and_read_project_file(input_file))`? It is now invoked unconditionally, I think.